File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.py
class Solution(object):
    def findDuplicates(self, nums):
        arr=[]
        obj={}
        for num in nums:
            if num in obj:
                obj[num]+=1
                if obj[num]==2:
                    arr.append(num)
            else:
                obj[num]=1
        return arr
        

File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDuplicates = function(nums) {
    let arr=[]
    let obj={}
    for(i=0;i<nums.length;i++){
        let num=nums[i]
        if(obj[num]){
            obj[num]++
            if(obj[num]==2){
                arr.push(num)
            }
        }else{
            obj[num]=1
        }
    }
    return arr
};